### PR TITLE
Add new context menu items

### PR DESCRIPTION
**Pull request by laurent22**
This pull request implements the change that was discussed [there](https://github.com/CalebJohn/joplin-rich-markdown/issues/22#issuecomment-930496482) and makes use of the new `joplin.workspace.filterEditorContextMenu` which allows dynamically setting items depending on the context. Using this and other new plugin APIs I've added the same menu items as in the preview - "Open link", "Reveal in folder" and "Copy path to clipboard" when right clicking on a resource.

I've also added "Toggle checkbox" for checkboxes.

The plugin requires the desktop app v2.7, which is not yet released!

I had to refactor the code a bit in ClickHandler to allow retrieving the type of content under the cursor. I think the code is equivalent but let me know if you prefer to do this differently.

Actual changes start [here](https://github.com/CalebJohn/joplin-rich-markdown/compare/main...laurent22:context-menu-patch?expand=1#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) (the rest is just an update to the plugin API)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -33,18 +37,91 @@ joplin.plugins.register({
...
+		// await joplin.commands.register({
+		// 	name: 'editor.richMarkdown.clickAtCursor',
+		// 	label: 'Perform action',
+		// 	iconName: 'fas fa-link',
+		// 	execute: async () => {
+		// 		await joplin.commands.execute('editor.execCommand', {
+		// 			name: 'clickUnderCursor',
+		// 		});
+		//   },
+		// });
```
Comment (CalebJohn) [inline src/index.ts:52]: This is still needed to bind Ctrl+Enter to the "Perform Action" command, which will follow a link or toggle a checkbox. It's a handy command that I use fairly often (mostly to toggle checkboxes).

**Comment by CalebJohn** [comment]
Thanks for the changes @laurent22! This is really cool, I saw the commits updating the plugin API for this the other day and I started scheming to add this myself. It's nice of you to beat me to it! 

I'm going to wait for 2.7 before merging this. I don't plan on making any changes to the plugin before then, so there shouldn't be any merge conflicts with this branch.

File: src/index.ts (hunk 1 above)
Comment (laurent22) [reply src/index.ts]: Right, I wasn't quite sure about this. I've restored the command now.

**Comment by CalebJohn** [comment]
Thanks again @laurent22!

**Pull request merged**